langlib/langlib-test/src/test/resources/test-src/xmllib_test.bal

```diff
@@ -62,6 +62,25 @@ function testConcat() returns xml {
     return 'xml:concat(x, <xml> testFromString(), "hello from String");
 }
 
+function testConcatWithXMLSequence() {
+    string a = "string one";
+    string b = "string two";
+
+    xml c = 'xml:concat(catalog, a, b);
```

Comment:
Shall we also assert here that the original `catalog` does not change?

Before calling `concat` we can create a clone, and then check `==` for that clone and `catalog` after calling `clone`.